### PR TITLE
read deploy credentials from disk

### LINKED ISSUE #269: Add sbt doc
http://alvinalexander.com/scala/how-to-generate-scala-documentation-scaladoc-command-examples

### PR DESCRIPTION
**Pull request by danyaljj**
- Now if the key is available inside `~/.ssh/` it never prompts the graphical username-pass window.   Hence we can use automatic deploys. 
- script to deploy the api-docs. Also a version is here: http://cogcomp.cs.illinois.edu/software/doc/saul/  --> Closes #269

### CODE REVIEW

**Comment by danyaljj** [comment]
Updated it with api-docs stuff.

**Comment by bhargav** [comment]
Looks good to me. Ok to merge.

File: build.sbt [hunk 1]
```diff
@@ -17,25 +17,36 @@ lazy val root = (project in file(".")).
...
     "ch.qos.logback" % "logback-classic" % "1.1.7"
   ),
   fork := true,
-  publishTo := Some(Resolver.sftp("CogcompSoftwareRepo", "bilbo.cs.illinois.edu", "/mounts/bilbo/disks/0/www/cogcomp/html/m2repo/")),
+  publishTo := Some(
+    Resolver.ssh(
+      "CogcompSoftwareRepo", "bilbo.cs.illinois.edu",
+      "/mounts/bilbo/disks/0/www/cogcomp/html/m2repo/").
+      as ("khashab2", new java.io.File(Path.userHome.absolutePath + "/.ssh/key"))
+  ),
```
Comment (kordjamshidi) [inline build.sbt]: I don't understand this piece of code, why there is a use name here? could you give me a link to some information to learn about it?

Comment (danyaljj) [inline build.sbt]: That's a good point. I know it is idea. In fast I tried that but did not work:
http://stackoverflow.com/questions/39261243/sbt-credentials-java-lang-illegalargumentexception-user-must-not-be-null
This is really the limitation of ivy.

**Comment by danyaljj** [comment]
Changed it so that it doesn't get contain the username inside the code; instead it gets it from the environment.

**Comment by kordjamshidi** [comment]
Thanks for figuring this out.  If semaphore is ok, feel free to merge.

**Pull request merged**